packages/framework/src/schemas/steps/channels/in-app.schema.ts

```diff
@@ -1,9 +1,69 @@
 import { Schema } from '../../../types/schema.types';
 
+const avatarSchema = {
+  type: 'object',
+  properties: {
+    type: { type: 'string', enum: ['user', 'system', 'custom'] },
+  },
+  oneOf: [
+    {
+      properties: {
+        type: { const: 'user' },
+        url: { type: 'string', format: 'uri' },
+      },
+      required: ['type', 'url'],
+      additionalProperties: false,
+    },
+    {
+      properties: {
+        type: { const: 'custom' },
+        url: { type: 'string', format: 'uri' },
+      },
+      required: ['type', 'url'],
+      additionalProperties: false,
+    },
+    {
+      properties: {
+        type: { const: 'system' },
+        icon: { type: 'string' },
+      },
+      required: ['type', 'icon'],
+      additionalProperties: false,
+    },
+  ],
+} as const satisfies Schema;
+
+const actionSchema = {
+  type: 'object',
+  properties: {
+    label: { type: 'string' },
+    url: { type: 'string' },
+  },
+  required: ['label'],
+  additionalProperties: false,
+} as const satisfies Schema;
+
+const actorSchema = {
+  type: 'object',
+  properties: {
+    subscriberId: { type: 'string' },
+    firstName: { type: 'string' },
```

Comment:
Actor won't always be a human. Would it make sense to concat first and last name into a displayName property and move control to the implementor to populate it correctly?

Reply:
I think such concatenation is best left to the presentation layer (i.e. to Inbox), folks might like to do things per custom inbox implementation. For example, take a subscriber with subset of `{ firstName: 'John', lastName: 'Doe' }`, presentation layer implementation might look like `displayName: ${firstName} ${lastName.chatAt(0)}.`-> `displayName: 'John D.'`.

I'd suggest we could introduce a `displayName` for `actor.type==='system'` to handle naming for a non-Human actor. That would come at a cost of a less unified API where both `displayName` and `firstName+lastName` must be handled in the Inbox for each `actor.type`

Reply:
There's also an argument that Framework can function as the presentation layer as Sok mentioned, such that `displayName` is constructed on the Framework and we have a unified `displayName` interface on the `actor`.

Reply:
We will defer `displayName` for now and revisit in a future iteraiton.